torch/autograd/profiler.py

```diff
@@ -339,37 +391,36 @@ def __missing__(self, key):
 ################################################################################
 # CPU checkpoints
 
-Record = namedtuple('Record', ['name', 'timestamp', 'kind'])
-
-
 def parse_cpu_trace(thread_records):
     next_id = 0
-    start_time = None
+    start_record = None
```

Comment:
This code assumes that `start_record` will be found as the first one, which might not always be the case (it's not guaranteed to be in the first list). This is why the code previously used two loops, to subtract the start time only after all events were processed.

Reply:
Why is it not guaranteed to be first? That deserves a comment at least.

Reply:
It is first in one of these lists, but it doesn't have to be in the first one. So far it happens to be like it, because it's the main Python thread and it gets registered as the first. However, if you were to run one profile in the main thread, then spawn another one and start profiling there, the start event wouldn't appear in the first list.